src/Components/Shared/src/ComponentParametersTypeCache.cs

```diff
@@ -40,10 +40,26 @@ internal sealed class ComponentParametersTypeCache
 
         if (assembly == null)
         {
-            return null;
+            // It might be that the assembly is not loaded yet, this can happen if the root component is defined in a
+            // different assembly than the app and there is no reference from the app assembly to any type in the class
+            // library that has been used yet.
+            // In this case, try and load the assembly and look up the type again.
+            // We only need to do this in the browser because its a different process, in the server the assembly will already
+            // be loaded.
+            if (OperatingSystem.IsBrowser())
+            {
+                try
+                {
+                    assembly = Assembly.Load(key.Assembly);
+                }
+                catch
+                {
+                    // It's fine to ignore the exception, since we'll return null below.
```

Comment:
Might the debugging experience be clearer if we allowed this to be unhandled? Then the developer will know the problem was with assembly loading, as opposed to some other issue with deserializing the parameter markers.

What is the error experience if we do ignore this exception?